lib/puppet/type/zypprepo.rb

```diff
@@ -1,298 +1,103 @@
-        `repositoryid` parameter in `zypper(8)`."
-      isnamevar
-    end
-
-    newproperty(:descr, parent: Puppet::IniProperty) do
-      desc "A human-readable description of the repository.
-        This corresponds to the name parameter in `zypper(8)`.
-        #{ABSENT_DOC}"
-      newvalue(:absent) { self.should = :absent }
-      newvalue(%r{.*}) {}
-      inikey 'name'
-    end
-
-    newproperty(:mirrorlist, parent: Puppet::IniProperty) do
-      desc "The URL that holds the list of mirrors for this repository.
-        #{ABSENT_DOC}"
-      newvalue(:absent) { self.should = :absent }
-      newvalue(%r{.*}) {}
-    end
+  newparam(:name) do
+    desc "The name of the repository.  This corresponds to the
+      `repositoryid` parameter in `zypper(8)`."
+    isnamevar
+  end
 
-    newproperty(:baseurl, parent: Puppet::IniProperty) do
-      desc "The URL for this repository. #{ABSENT_DOC}"
-      newvalue(:absent) { self.should = :absent }
-      # Should really check that it's a valid URL
-      newvalue(%r{.*}) {}
-    end
+  newproperty(:descr) do
+    desc "A human-readable description of the repository.
+      This corresponds to the name parameter in `zypper(8)`.
+      #{ABSENT_DOC}"
+    newvalue(:absent) { self.should = :absent }
+    newvalue(%r{.*}) {}
+  end
 
-    newproperty(:path, parent: Puppet::IniProperty) do
-      desc "The path relative to the baseurl. #{ABSENT_DOC}"
-      newvalue(:absent) { self.should = :absent }
-      newvalue(%r{.*}) {}
-    end
+  newproperty(:mirrorlist) do
+    desc "The URL that holds the list of mirrors for this repository.
+      #{ABSENT_DOC}"
+    newvalue(:absent) { self.should = :absent }
+    newvalue(%r{.*}) {}
+  end
 
-    newproperty(:enabled, parent: Puppet::IniProperty) do
-      desc "Whether this repository is enabled, as represented by a
-        `0` or `1`. #{ABSENT_DOC}"
-      newvalue(:absent) { self.should = :absent }
-      newvalue(%r{(0|1)}) {}
-    end
+  newproperty(:baseurl) do
+    desc "The URL for this repository. #{ABSENT_DOC}"
+    newvalue(:absent) { self.should = :absent }
+    # Should really check that it's a valid URL
+    newvalue(%r{.*}) {}
+  end
 
-    newproperty(:gpgcheck, parent: Puppet::IniProperty) do
-      desc "Whether to check the GPG signatures
-        from this repository, as represented by a `0` or `1`.
-        #{ABSENT_DOC}"
-      newvalue(:absent) { self.should = :absent }
-      newvalue(%r{(0|1)}) {}
-    end
+  newproperty(:path) do
+    desc "The path relative to the baseurl. #{ABSENT_DOC}"
+    newvalue(:absent) { self.should = :absent }
+    newvalue(%r{.*}) {}
+  end
 
-    newproperty(:repo_gpgcheck, parent: Puppet::IniProperty) do
-      desc "Whether to check the GPG signature on the repository
-         metadata, as represented by a `0` or `1`.
-        #{ABSENT_DOC}"
-      newvalue(:absent) { self.should = :absent }
-      newvalue(%r{(0|1)}) {}
-    end
```

Comment:
@tuxmea This reverted #48. Was it an overlook or is there any reason why it was reverted?

Reply:
seems as if I overlooked two properties. going to update with the actual open PR

Reply:
🙇 Thanks!

Reply:
readded properties to #55 
At the moment I only have one more issue: correcting e.g. priority is not working.